diff --git a/src/gmpv_controller.c b/src/gmpv_controller.c
index e259e85..76ca860 100644
--- a/src/gmpv_controller.c
+++ b/src/gmpv_controller.c
@@ -860,10 +860,12 @@ static void fullscreen_handler(		GObject *object,
 	GActionMap *map = G_ACTION_MAP(window);
 	GAction *toggle_playlist = NULL;
 	GAction *toggle_controls = NULL;
+	GAction *set_video_size  = NULL;
 	gboolean fullscreen = FALSE;
 
 	toggle_playlist = g_action_map_lookup_action(map, "toggle-playlist");
 	toggle_controls = g_action_map_lookup_action(map, "toggle-controls");
+	set_video_size  = g_action_map_lookup_action(map, "set-video-size");
 
 	g_object_get(view, "fullscreen", &fullscreen, NULL);
 
@@ -871,6 +873,8 @@ static void fullscreen_handler(		GObject *object,
 		(G_SIMPLE_ACTION(toggle_playlist), !fullscreen);
 	g_simple_action_set_enabled
 		(G_SIMPLE_ACTION(toggle_controls), !fullscreen);
+	g_simple_action_set_enabled
+		(G_SIMPLE_ACTION(set_video_size), !fullscreen);
 }
 
 static void play_button_handler(GtkButton *button, gpointer data)
